internal/internal_schedule_client.go

```diff
@@ -0,0 +1,789 @@
+	rangesPB := make([]*schedulepb.Range, len(scheduleRange))
+	for i, r := range scheduleRange {
+		rangesPB[i] = &schedulepb.Range{
+			Start: int32(r.Start),
+			End:   int32(r.End),
+			Step:  int32(r.Step),
+		}
+	}
+	return rangesPB
+}
+
+func convertFromPBRangeList(scheduleRangePB []*schedulepb.Range) []ScheduleRange {
+	scheduleRange := make([]ScheduleRange, len(scheduleRangePB))
+	for i, r := range scheduleRangePB {
+		if r == nil {
+			continue
+		}
+		scheduleRange[i] = ScheduleRange{
+			Start: int(r.Start),
+			End:   int(r.End),
+			Step:  int(r.Step),
+		}
+	}
+	return scheduleRange
+}
+
+func convertFromPBScheduleCalendarSpecList(calendarSpecPB []*schedulepb.StructuredCalendarSpec) []ScheduleCalendarSpec {
+	calendarSpec := make([]ScheduleCalendarSpec, len(calendarSpecPB))
+	for i, e := range calendarSpecPB {
+		calendarSpec[i] = ScheduleCalendarSpec{
+			Second:     convertFromPBRangeList(e.Second),
+			Minute:     convertFromPBRangeList(e.Minute),
+			Hour:       convertFromPBRangeList(e.Hour),
+			DayOfMonth: convertFromPBRangeList(e.DayOfMonth),
+			Month:      convertFromPBRangeList(e.Month),
+			Year:       convertFromPBRangeList(e.Year),
+			DayOfWeek:  convertFromPBRangeList(e.DayOfWeek),
+			Comment:    e.Comment,
+		}
+	}
+	return calendarSpec
+}
+
+func convertToPBScheduleCalendarSpecList(calendarSpec []ScheduleCalendarSpec) []*schedulepb.StructuredCalendarSpec {
+	calendarSpecPB := make([]*schedulepb.StructuredCalendarSpec, len(calendarSpec))
+	for i, e := range calendarSpec {
+		calendarSpecPB[i] = &schedulepb.StructuredCalendarSpec{
+			Second:     convertToPBRangeList(e.Second),
+			Minute:     convertToPBRangeList(e.Minute),
+			Hour:       convertToPBRangeList(e.Hour),
+			DayOfMonth: convertToPBRangeList(e.DayOfMonth),
+			Month:      convertToPBRangeList(e.Month),
+			Year:       convertToPBRangeList(e.Year),
+			DayOfWeek:  convertToPBRangeList(e.DayOfWeek),
+			Comment:    e.Comment,
+		}
+	}
+	return calendarSpecPB
+}
+
+func convertFromPBScheduleActionResultList(aa []*schedulepb.ScheduleActionResult) []ScheduleActionResult {
+	recentActions := make([]ScheduleActionResult, len(aa))
+	for i, a := range aa {
+		var workflowExecution *ScheduleWorkflowExecution
+		if a.GetStartWorkflowResult() != nil {
+			workflowExecution = &ScheduleWorkflowExecution{
+				WorkflowID:          a.GetStartWorkflowResult().GetWorkflowId(),
+				FirstExecutionRunID: a.GetStartWorkflowResult().GetRunId(),
+			}
+		}
+		recentActions[i] = ScheduleActionResult{
+			ScheduleTime:        common.TimeValue(a.GetScheduleTime()),
+			ActualTime:          common.TimeValue(a.GetActualTime()),
+			StartWorkflowResult: workflowExecution,
+		}
+	}
+	return recentActions
+}
+
+func encodeScheduleWorklowArgs(dc converter.DataConverter, args []interface{}) (*commonpb.Payloads, error) {
+	payloads := make([]*commonpb.Payload, len(args))
+	for i, arg := range args {
+		// arg is already encoded
```

Comment:
I support a mix of serialized and not serialized Args. I found it was nice when writing integration tests.

Reply:
This is probably ok, but technically this is a bit different than elsewhere where we'd call `DataConverter.ToPayloads` (plural) w/ all args if they wanted to, say, condense them or something. But most people treat that as the same as calling each individually.

You may want to document on args, search attributes, and memos that we just pass through payloads on create/update.